src/python/pants/backend/jvm/tasks/nailgun_task.py

```diff
@@ -49,17 +57,30 @@ def __init__(self, *args, **kwargs):
 
     id_tuple = (self.ID_PREFIX, self.__class__.__name__)
 
+    self._execution_strategy = self.set_execution_strategy()
     self._identity = '_'.join(id_tuple)
     self._executor_workdir = os.path.join(self.context.options.for_global_scope().pants_workdir,
                                           *id_tuple)
-
+  
+  def set_execution_strategy(self):
```

Comment:
Oh, one last thing! This should probably be `execution_strategy` or `get_execution_strategy`, not `set_`! :)